### PR TITLE
PXP-8703 Updated process filename logic

### PR DESCRIPTION
**Pull request by rnerella92**
When a file is uploaded using gen3-client with --include-subdirectory flag on then the filename is empty from indexd record. 

for example: 
- ./gen3-client upload --profile=ramudev --include-subdirname=true --upload-path=/Users/ramunerella/Desktop/test [filename is not empty]
- ./gen3-client upload --profile=ramudev --include-subdirname=true --upload-path=/Users/ramunerella/Desktop/test/test_subfolder/test020.dcm [filename is empty]

- Updated process filename logic

### Bug Fixes
Updated filename name logic when 'includesubdir' flag is on.

### CODE REVIEW

File: gen3-client/g3cmd/utils.go [hunk 1]
```diff
@@ -467,7 +467,22 @@ func ProcessFilename(uploadPath string, filePath string, includeSubDirName bool,
...
+				fmt.Println("uploadpath is not valid")
+			}
+			if !fileInfo.IsDir() {
+				_, file := filepath.Split(uploadPath)
+				presentDirname = strings.TrimSuffix(uploadPath, file)
+
+			} else {
+				presentDirname = strings.TrimSuffix(uploadPath, commonUtils.PathSeparator+"*")
+
+			}
```
Comment (m0nhawk) [inline gen3-client/g3cmd/utils.go:484]:
```suggestion
			if !fileInfo.IsDir() {
				presentDirname, _ := filepath.Split(uploadPath)
			} else {
				presentDirname = strings.TrimSuffix(uploadPath, commonUtils.PathSeparator+"*")
			}
```

Comment (m0nhawk) [reply gen3-client/g3cmd/utils.go:484]: This can be done simpler.

**Review by m0nhawk — CHANGES_REQUESTED** [review]

File: gen3-client/g3cmd/utils.go [hunk 2]
```diff
@@ -467,7 +467,21 @@ func ProcessFilename(uploadPath string, filePath string, includeSubDirName bool,
...
+		if err == nil {
+			fileInfo, err := os.Stat(uploadPath)
+			if os.IsNotExist(err) {
+				fmt.Println("uploadpath is not valid")
+			}
+			if !fileInfo.IsDir() {
+				presentDirname, _ = filepath.Split(uploadPath)
+
+			}
+			if fileInfo.IsDir() {
```
Comment (m0nhawk) [inline gen3-client/g3cmd/utils.go:481]:
```suggestion
			if !fileInfo.IsFile() {
```

**Review by m0nhawk — APPROVED** [review]

File: gen3-client/g3cmd/utils.go [hunk 3]
```diff
@@ -467,7 +467,21 @@ func ProcessFilename(uploadPath string, filePath string, includeSubDirName bool,
...
+				presentDirname = strings.TrimSuffix(uploadPath, commonUtils.PathSeparator+"*")
+			}
```
Comment (m0nhawk) [inline gen3-client/g3cmd/utils.go:483]:
```suggestion
			if !fileInfo.IsDir() {
				presentDirname, _ = filepath.Split(uploadPath)
			} else {
				presentDirname = strings.TrimSuffix(uploadPath, commonUtils.PathSeparator+"*")
			}
```

**Review by m0nhawk — APPROVED** [review]

File: gen3-client/g3cmd/utils.go [hunk 4]
```diff
@@ -466,8 +466,24 @@ func ProcessFilename(uploadPath string, filePath string, includeSubDirName bool,
...
+				presentDirname = strings.TrimSuffix(uploadPath, commonUtils.PathSeparator+"*")
+
+			} else {
+				pwd, err := os.Getwd()
+				if err != nil {
+					log.Fatal(err)
+				}
+				presentDirname = pwd
+			}
+		}
```
Comment (m0nhawk) [inline gen3-client/g3cmd/utils.go:486]:
```suggestion
		uploadPath, err := commonUtils.GetAbsolutePath(uploadPath)
		if err == nil {
			presentPath := strings.TrimSuffix(uploadPath, commonUtils.PathSeparator + "*")
			fileInfo, err := os.Stat(presentPath)
			if err != nil {
				log.Fatal(err)
			}

			if !fileInfo.IsDir() {
				// do nothing (explicitly)
			} else {
				filename = strings.TrimPrefix(filePath, presentPath)
				filename = strings.TrimPrefix(filename, commonUtils.PathSeparator)
			}
		}
```

**Review by m0nhawk — CHANGES_REQUESTED** [review]

**Review by m0nhawk — APPROVED** [review]
